nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,311 @@
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.ProcessorInitializationContext;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.InputStreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.util.ObjectHolder;
+
+import org.apache.tika.exception.TikaException;
+import org.apache.tika.io.TikaInputStream;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.sax.BodyContentHandler;
+import org.xml.sax.SAXException;
+
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@Tags({"media", "file", "format", "metadata", "audio", "video", "image", "document", "pdf"})
+@CapabilityDescription("Extract the content metadata from flowfiles containing audio, video, image, and other file "
+        + "types.  This processor relies on the Apache Tika project for file format detection and parsing.  It "
+        + "extracts a long list of metadata types for media files including audio, video, and print media "
+        + "formats."
+        + "For the more details and the list of supported file types, visit the library's website "
+        + "at http://tika.apache.org/.")
+@WritesAttributes({@WritesAttribute(attribute = "<Metadata Key Prefix>.<attribute>", description = "The extracted content metadata "
+        + "will be inserted with the attribute name \"<Metadata Key Prefix>.<attribute>\", or \"<attribute>\" if "
+        + "\"Metadata Key Prefix\" is not provided.")})
+@SupportsBatching
+public class ExtractMediaMetadata extends AbstractProcessor {
+
+    static final PropertyDescriptor MAX_NUMBER_OF_ATTRIBUTES = new PropertyDescriptor.Builder()
+            .name("Max Number of Attributes")
+            .description("Specify the max number of attributes to add to the flowfile. There is no guarantee in what order"
+                    + " the tags will be processed. By default it will process all of them.")
+            .required(false)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    private static final PropertyDescriptor MAX_ATTRIBUTE_LENGTH = new PropertyDescriptor.Builder()
+            .name("Max Attribute Length")
+            .description("Specifies the maximum length of a single attribute value.  When a metadata item has multiple"
+                    + " values, they will be merged until this length is reached and then \", ...\" will be added as"
+                    + " an indicator that additional values where dropped.  If a single value is longer than this, it"
+                    + " will be truncated and \"(truncated)\" appended to indicate that truncation occurred.")
+            .required(true)
+            .defaultValue("10240")
```

Comment:
So, I changed the default to 100, since the typical "X-Parsed-By" value usually includes the DefaultParser and a format specific parser which together are more than 50 since fully qualified names are used.